### PR TITLE
Improve documentation on decoupled backend API

### PR DESCRIPTION
**Pull request by tanmayv25**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -447,13 +449,73 @@ For a batch of requests the backend should attempt to combine the
...
 request or not send any responses for a request. A backend may also
 send responses out-of-order relative to the order that the request
-batches are executed. Backends and models that operate in this way are
-referred to as *decoupled* backends and models, and are typically much
-more difficult to implement. The [repeat example](examples/README.md)
-shows a simplified implementation of a decoupled backend.
+batches are executed. Such backends are called *decoupled* backends.
+The decoupled backends use `ResponseFactory` object per request to keep
+create and send any number of responses.  For this kind of backend
+executing a single inference requests typically requires the following
```
Comment (nv-kmcgill53) [inline README.md:460]: "executing a single inference request typically"

File: README.md [hunk 2]
```diff
@@ -447,13 +449,73 @@ For a batch of requests the backend should attempt to combine the
...
+  containing the tensor contents.
+
+* Create a `ResponseFactory` object for the request using
+  TRITONBACKEND_ResponseFactoryNew.
+
+  1. Create a response from the `ResponseFactory` object using
+  TRITONBACKEND_ResponseNewFromFactory. As long as you have
+  `ResponseFactory` object you can continue creating responses.
+
+  2. For each output tensor that the request expects to be returned, use
```
Comment (nv-kmcgill53) [inline README.md:474]: output tensor which the request expects to be returned

**Review by nv-kmcgill53 — APPROVED** [review]

**Review by GuanLuo — DISMISSED** [review]

**Review by nv-kmcgill53 — APPROVED** [review]

**Review by Tabrizian — DISMISSED** [review]

File: README.md [hunk 3]
```diff
@@ -418,8 +418,10 @@ or access them in any way. For more detailed description of request/response
 lifetimes, study the documentation of TRITONBACKEND_ModelInstanceExecute in
 [tritonbackend.h](https://github.com/triton-inference-server/core/blob/main/include/triton/core/tritonbackend.h).
 
+##### Single Response
+
 Most backends will create a single response for each request. For that
-kind of backend executing a single inference requests requires the
+kind of backend executing a single inference request requires the
```
Comment (CoderHam) [inline README.md:424]: nit: backend, executing

File: README.md [hunk 4]
```diff
@@ -447,13 +449,76 @@ For a batch of requests the backend should attempt to combine the
...
+
 It is also possible for a backend to send multiple responses for a
 request or not send any responses for a request. A backend may also
 send responses out-of-order relative to the order that the request
-batches are executed. Backends and models that operate in this way are
-referred to as *decoupled* backends and models, and are typically much
-more difficult to implement. The [repeat example](examples/README.md)
-shows a simplified implementation of a decoupled backend.
+batches are executed. Such backends are called *decoupled* backends.
+The decoupled backends use `ResponseFactory` object per request to keep
```
Comment (CoderHam) [inline README.md:458]: use one `ResponseFactory` object per request

File: README.md [hunk 5]
```diff
@@ -447,13 +449,76 @@ For a batch of requests the backend should attempt to combine the
...
+creating and sending any number of responses for the request.  For this
```
Comment (CoderHam) [inline README.md:459]: Remove extra space before "For"

**Review by CoderHam — CHANGES_REQUESTED** [review]

**Review by CoderHam — APPROVED** [review]

**Pull request merged**